### PR TITLE
Support Windows on Azure Pipelines [continued...]

### PR DESCRIPTION
**Pull request by joerick**
Hello, I had a chance to look at Azure pipelines support again, building on @astrofrog 's work in #126.

Changes I've made:

- Vendors appveyor_run_with_cmd, but disable it on azure pipelines
- Adds azure-pipelines test run config
- Adds auto-detect platform for Azure

Questions-

- I removed the run_with_env script, since it was Appveyor-specific. But I'm not sure if something similar is still needed on Azure Pipelines... I'm thinking about the lines:

```
ECHO Configuring Windows SDK %WINDOWS_SDK_VERSION% for Python %MAJOR_PYTHON_VERSION% on a 64 bit architecture
        SET DISTUTILS_USE_SDK=1
        SET MSSdk=1
        "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
        "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
```

Those env vars are only set on 64-bit Python 2.7... which now has a separate toolchain. Maybe it's fine.

I'm now getting green builds over at https://github.com/joerick/cibuildwheel-azure-example :)

Still to do:

- [x] Exclude Python 3.4 from tests, when running on Azure
- [x] Write some docs, with a minimal setup example ~(maybe using the config template feature)~

### CODE REVIEW

**Comment by astrofrog** [comment]
Very nice! 👍

File: README.md [hunk 1]
```diff
@@ -33,6 +33,69 @@ Usage
...
+        cibuildwheel --output-dir wheelhouse .
+    - task: PublishBuildArtifacts@1
+      inputs: {pathtoPublish: 'wheelhouse'}
+- job: windows
+  pool: {vmImage: 'vs2017-win2016'}
+  steps: 
+    - {task: UsePythonVersion@0, inputs: {versionSpec: '2.7', architecture: x86}}
+    - {task: UsePythonVersion@0, inputs: {versionSpec: '2.7', architecture: x64}}
+    - {task: UsePythonVersion@0, inputs: {versionSpec: '3.4', architecture: x86}}
+    - {task: UsePythonVersion@0, inputs: {versionSpec: '3.4', architecture: x64}}
```
Comment (astrofrog) [inline README.md]: Should 3.4 be removed here since it won't be supported?

Comment (astrofrog) [reply README.md]: Although I guess it's fine for pure-Python packages, so maybe fine to leave it.

Comment (mayeut) [reply README.md]: Python 3.4 has reached End-Of-Life, it can probably be removed.

Comment (joerick) [reply README.md]: I've removed this from the Azure config. We can remove support for 3.4 from the entire cibuildwheel at some point, but as long as it's working I'm happy to leave it in for now.

**Comment by joerick** [comment]
Great! I think this is ready.

I've abandoned the config template reference bit because I was getting strange errors like `Repository cibuildwheel requires an endpoint Parameter name: endpoint` and I didn't want it to hold up releasing this.

Comments/review welcome.

**Comment by astrofrog** [comment]
This looks good to me! :+1: Looking forward to using it 😄

**Pull request merged**